ria_remote/remote.py

```diff
@@ -353,7 +441,85 @@ def _local_io(self):
         #return self.objtree_base_path.is_dir()
         return not self.storage_host
 
+    def _check_layout_version(self):
+        """Check whether we can deal with the layout reported by the remote end
+
+        There are two aspects of layout versioning:
+        - the tree to put the datasets in (version recorded in base_path/ria-layout-version)
+        - the tree of the actual annex objects of a particular dataset (version recorded in
+          dataset_somewhere_beneath_base_path/ria-layout-version)
+
+        If the version found on the remote end isn't supported and `force-write` isn't configured,
+        this sets the remote to read-only operation.
+        """
+
+        dataset_tree_version_file = \
+            self.objtree_base_path / 'ria-layout-version'
+        object_tree_version_file = \
+            self.objtree_base_path / self.archive_id[:3] / self.archive_id[3:] / 'ria-layout-version'
+
+        # TODO: It might be faster to directly try to read it, parse the output to detect non-existence of the file
+        #       and act upon it, rather than having to separate remote calls executed for checking existence and then
+        #       read the content
+
+        # 1. check dataset tree version
+        if not self.io.exists(dataset_tree_version_file):
+            # we are first, just put our stamp on it
+            self.io.mkdir(dataset_tree_version_file.parent)
+            self.io.write_file(dataset_tree_version_file, self._dataset_tree_version)
+            # TODO: Catch (and parse?) possible errors to know when we don't need to care.
+            #       We might not have permission to write to the base path. This doesn't necessarily prevent us from
+            #       doing everything else, so ignore and let someone else care for it
```

Comment:
I don't think this OK. We simply have no knowledge about this path. It would be OK, if there was no directory at all.

Reply:
Just to clarify: If there's nothing at all, then we can write to it. If there is but no version file is present, we don't put it there but set read_only?
I tend to agree, although this would mean that some dedicated initial setup routine is needed. But where would that go? We certainly don't want to be the exception that requires `initremote` to actually have access. Dedicated script in `bin/`?